### PR TITLE
New package: iasl-20180105

### PR DESCRIPTION
**Pull request by tim241**

### CODE REVIEW

File: srcpkgs/iasl/template [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Template file for 'iasl'
+pkgname=iasl
+version=20180105
+revision=1
+wrksrc="acpica-unix-$version"
+only_for_archs="i686 x86_64"
+build_style=gnu-configure
+make_build_args="iasl"
+hostmakedepends="gcc flex bison"
+makedepends="gcc flex bison"
```
Comment (maxice8) [inline srcpkgs/iasl/template]: There is no need to mark GCC as a dependency it is automatically in

File: srcpkgs/iasl/template [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Template file for 'iasl'
+pkgname=iasl
+version=20180105
+revision=1
+wrksrc="acpica-unix-$version"
+only_for_archs="i686 x86_64"
```
Comment (maxice8) [inline srcpkgs/iasl/template]: Please add a comment with the reason why making this exclusive to those arches ( something brief like,  `only arches supported by upstream` )

File: srcpkgs/iasl/template [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+hostmakedepends="gcc flex bison"
+makedepends="gcc flex bison"
+short_desc="An ACPI tool"
+maintainer="Tim Wanders <timwanders241@gmail.com>"
+license="GPL-2,Intel"
+homepage="https://www.acpica.org"
+distfiles="https://acpica.org/sites/acpica/files/acpica-unix-$version.tar.gz"
+checksum=414f843ac6c7c53bbd2a830b092a2a27c49172b0878fd27fe306dd526b78b921
+do_configure() {
+	make clean
```
Comment (maxice8) [inline srcpkgs/iasl/template]: This looks weird, why override the configure phase ?

File: srcpkgs/iasl/template [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+revision=1
+wrksrc="acpica-unix-$version"
+only_for_archs="i686 x86_64"
+build_style=gnu-configure
+make_build_args="iasl"
+hostmakedepends="gcc flex bison"
+makedepends="gcc flex bison"
+short_desc="An ACPI tool"
+maintainer="Tim Wanders <timwanders241@gmail.com>"
+license="GPL-2,Intel"
```
Comment (maxice8) [inline srcpkgs/iasl/template]: Style: add a space between licenses, if the license isn't found in /usr/share/licenses please add a vlicense.

File: srcpkgs/iasl/template (hunk 3 above)
Comment (tim241) [reply srcpkgs/iasl/template]: There is no `configure` file, only a `Makefile`

Comment (maxice8) [reply srcpkgs/iasl/template]: Then the build_style should be gnu-makefile

File: srcpkgs/iasl/template (hunk 4 above)
Comment (tim241) [reply srcpkgs/iasl/template]: There is no `LICENSE` file, only on their site, there is license information about the source weirdly enough.
More info here: https://acpica.org/Licensing

File: srcpkgs/iasl/template (hunk 1 above)
Comment (cr6git) [reply srcpkgs/iasl/template]: `makedepends="gcc flex bison"`
What is that??
These are `hostmakedepends`, not `makedepends` !

Comment (tim241) [reply srcpkgs/iasl/template]: okay, did not know that, thanks!

File: srcpkgs/iasl/template (hunk 4 above)
Comment (maxice8) [reply srcpkgs/iasl/template]: I'd just pick the GPL-2 license, the licensing fluff gives us the option to pick any of the three, the BSD ones require we also ship a LICENSE File.

Comment (tim241) [reply srcpkgs/iasl/template]: okay

File: srcpkgs/iasl/template (hunk 3 above)
Comment (cr6git) [reply srcpkgs/iasl/template]: It's the same on the [PKGBUILD](https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/iasl#n26), however.

Comment (tim241) [reply srcpkgs/iasl/template]: okay

Comment (maxice8) [reply srcpkgs/iasl/template]: @cr6git yeah but that should be either pre_build() or post_extract()

Comment (cr6git) [reply srcpkgs/iasl/template]: @maxice8 oh sorry, I should have reload the page in my browser, in order to see your last comment. 😉

File: srcpkgs/iasl/template [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+maintainer="Tim Wanders <timwanders241@gmail.com>"
+license="GPL-2"
+homepage="https://www.acpica.org"
+distfiles="https://acpica.org/sites/acpica/files/acpica-unix-$version.tar.gz"
+checksum=414f843ac6c7c53bbd2a830b092a2a27c49172b0878fd27fe306dd526b78b921
+pre_build() {
+	make clean
+}
+do_build() {
+	make
```
Comment (maxice8) [inline srcpkgs/iasl/template]: If your build_style is gnu-makefile then do_build already does make for you.

Comment (tim241) [reply srcpkgs/iasl/template]: if I try to remove that, the build breaks, this resolves the build issue

Comment (maxice8) [reply srcpkgs/iasl/template]: log ? it shouldn't break unless some rogue variable is set.

Comment (tim241) [reply srcpkgs/iasl/template]: here's the build without the `do_build` part: https://gist.github.com/3fee6425e36fcb139cc3c0aa6144e4eb

**Comment by maxice8** [comment]
On Sat, Feb 03, 2018 at 03:36:37AM -0800, Tim Wanders wrote:
> 
> here's the build without the `do_build` part: https://gist.github.com/3fee6425e36fcb139cc3c0aa6144e4eb

Taking a look into it

File: srcpkgs/iasl/template (hunk 5 above)
Comment (tim241) [reply srcpkgs/iasl/template]: and here's the build with the `do_build` part: https://gist.github.com/10384511ef35d0f6aa5acb73fed48b48

Comment (maxice8) [reply srcpkgs/iasl/template]: yeah, looks fine to me.

**Comment by cr6git** [comment]
@maxice8 Probably needs `pciutils-devel` ?

**Comment by tim241** [comment]
I didn't see acpica-utils, so this package is unrequired, closing!

**Pull request closed without merging**